ArduCopter/mode_guided.cpp

```diff
@@ -436,29 +475,44 @@ void ModeGuided::pos_control_run()
     // set motors to full range
     motors->set_desired_spool_state(AP_Motors::DesiredSpoolState::THROTTLE_UNLIMITED);
 
-    // run waypoint controller
-    copter.failsafe_terrain_set_status(wp_nav->update_wpnav());
+    bool do_avoid = false;
+    guided_vel_target_cms = pos_control->get_vel_target_cms();
+#if AC_AVOID_ENABLED
+    // limit the velocity for obstacle/fence avoidance
+    copter.avoid.adjust_velocity(guided_vel_target_cms, pos_control->get_pos_xy_p().kP(), pos_control->get_max_accel_xy_cmss(), pos_control->get_pos_z_p().kP(), pos_control->get_max_accel_z_cmss(), G_Dt);
+    do_avoid = copter.avoid.limits_active();
+#endif
+
+    // send position and velocity targets to position controller
+    guided_accel_target_cmss.zero();
+    if (do_avoid) {
```

Comment:
I wonder what would happen if the vehicle was flying quickly to the west and the user then input a position target to the east.  Directly west of the vehicle was a fence and BendyRuler was active.  I think what would happen is the position controller init would set the target velocity to the vehicle's initial velocity.  Then this would be passed into avoid.adjust_velocity() (from within pos_control_run()).  Avoidance would shorten the desired velocity so that the vehicle doesn't hit the wall .. i guess it would be shortened and shortened and hopefully would eventually get to zero and "do_avoid" would become false and then the position controller would finally get a chance to send the vehicle in the correct direction.... so I guess it should work but I just wanted to point out that the velocity target might not have come from the current position.  I suppose a similar situation could happen if the external caller updates the position target.

Reply:
Removed it.